kroxylicious/example-proxy-config.yml

```diff
@@ -10,6 +10,7 @@ adminHttp:
     prometheus: {}
 virtualClusters:
   demo:
+    name: "Demo Cluster"
```

Comment:
Why do we need both the key to the map `virtualClusters` and name in the configuration model?  Should the map be changed into a collection  and we then have a business rule that clusters need to have a distinct name?

Reply:
Hmm good question. 

My thoughts were:
- Less invasive
- Backwards compatible 
- Leaves users more flexibility for human readable names 


Of those the the backwards compatibility is probably the strongest, but none are great on their own

Reply:
One other advantage I thinking about it using a map leads to a more natural layout in the config file. With a map clusters are naturally scoped to by the name.

Reply:
We discussed this on [slack](https://kroxylicious.slack.com/archives/C04V1K6EAKZ/p1690834640354339) the TL;DR version is
- the natural scoping is good 
- if we enable [STRICT_DUPLICATE_DETECTION](https://fasterxml.github.io/jackson-core/javadoc/2.9/com/fasterxml/jackson/core/JsonParser.Feature.html#STRICT_DUPLICATE_DETECTION) we can detect duplicates. 
- We don't have any immediate need for display names
- Making configuration problems fatal is a nice property and we can still support re-loading config at some future time

Reply:
Enabled `STRICT_DUPLICATE_DETECTION` in 0e868ab84160892c8dcfb328713cf993f61d826d 
and dropped `clusterName` as a config property in cb2faf7a9cb407a92dce6130e3ca17e1ef19e8c5